disnake/ext/commands/base_core.py

```diff
@@ -202,6 +202,8 @@ def _ensure_assignment_on_copy(self, other: AppCommandT) -> AppCommandT:
         if self._max_concurrency != other._max_concurrency:
             # _max_concurrency won't be None at this point
             other._max_concurrency = cast(MaxConcurrency, self._max_concurrency).copy()
+        if self.body._default_member_permissions != other.body._default_member_permissions:
```

Comment:
Think this would have to be something like this:
```suggestion
        if (
            self.body._default_member_permissions != other.body._default_member_permissions
            and "default_member_permissions" not in other.__original_kwargs__
        ):
```

Otherwise this would remove `default_member_permissions` set through `slash_command_attrs`.

However, this results in some inconsistencies when setting `default_member_permissions` in `slash_command_attrs` *and* through the decorator:
```py
class Misc(commands.Cog, slash_command_attrs={"default_member_permissions": 64}):
    @commands.default_member_permissions(32)
    @commands.slash_command()
    async def bugged(self, _):
        ...

    @commands.slash_command()
    # @commands.default_member_permissions(32)
    async def workaround(self, _):
        ...
```
In the above example, both commands end up with `default_member_permissions=64` - uncommenting the second decorator results in an exception, the first one seems fine and just gets silently overridden by the `slash_command_attrs` value.

Reply:
@EQUENOS

Reply:
https://github.com/DisnakeDev/disnake/pull/678/commits/494d1af8395b04118b8d999680f10569ab17b165